daemon/logger/logger.go

```diff
@@ -57,7 +58,7 @@ type LogWatcher struct {
 	Msg chan *Message
 	// For sending error messages that occur while while reading logs.
 	Err           chan error
-	closeNotifier chan struct{}
+	closeNotifier *pubsub.Publisher
```

Comment:
Otherwise I'd either have to store some state about the LogWatcher having already been closed and use a mutex, or essentially implement what the publisher's Close function does.
I can change, this just made it super tiny.

The reason for the change is I found a race while testing this issue where by the client has closed so Close is called, and then called `docker rm -f` on the container, and Close was called again because the reader was not yet evicted.